relay-event-normalization/src/event.rs

```diff
@@ -2321,4 +2321,53 @@ mod tests {
         }
         "###);
     }
+
+    #[test]
+    fn test_accept_recent_transactions_with_stale_timestamps() {
+        let config = NormalizationConfig {
+            received_at: Some(Utc::now()),
+            max_secs_in_past: Some(2),
+            max_secs_in_future: Some(1),
+            ..Default::default()
+        };
+
+        let json = r#"{
+  "event_id": "52df9022835246eeb317dbd739ccd059",
+  "transaction": "i have an stale timestamp, but im recent!",
+  "start_timestamp": -2,
+  "timestamp": -1
+}"#;
+        let mut event = Annotated::<Event>::from_json(json).unwrap();
+
+        assert!(normalize_event(&mut event, &config).is_ok());
+    }
+
+    #[test]
+    fn test_reject_stale_transactions_after_timestamp_normalization() {
+        let now = Utc::now();
+        let config = NormalizationConfig {
+            received_at: Some(now),
+            max_secs_in_past: Some(2),
+            max_secs_in_future: Some(1),
+            ..Default::default()
+        };
+
+        let json = format!(
+            r#"{{
+          "event_id": "52df9022835246eeb317dbd739ccd059",
+          "transaction": "clockdrift is not enough to accept me :(",
+          "start_timestamp": -62135811111,
```

Comment:
Could you explain why this test fails and the other one passes? Both contain timestamps that are extremely out of date.

Maybe it also helps to assert on the corrected timestamp in the passing test.